src/lib/RelayOrderLib.sol

```diff
@@ -37,10 +37,6 @@ library RelayOrderLib {
             revert ReactorErrors.DeadlineBeforeEndTime();
         }
 
-        if (block.timestamp > order.info.deadline) {
-            revert ReactorErrors.DeadlinePassed();
-        }
```

Comment:
Wait why are we removing this?

Reply:
Are we pushing this to the P2 transfer? I still think we need this since there can be orders with no inputs (sponsored txn maybe) and they need deadlines.

Reply:
this will still be checked if there are no inputs bc we need to still call p2 for the sig verification